app/components/offer-filters-button.js

```diff
@@ -0,0 +1,50 @@
+import Ember from "ember";
+import _ from "lodash";
+
+export default Ember.Component.extend({
+  filterService: Ember.inject.service(),
+
+  offerStateFilters: Ember.computed.alias("filterService.offerStateFilters"),
+  hasStateFilters: Ember.computed("offerStateFilters", function() {
+    return this.get("offerStateFilters").length > 0;
+  }),
+
+  offerTimeRange: Ember.computed.alias("filterService.offerTimeRange"),
+
+  selfReviewFilter: Ember.computed.alias("filterService.selfReviewFilter"),
+
+  hasTimeFilters: Ember.computed("offerTimeRange", function() {
+    const { preset, after, before } = this.get("offerTimeRange");
+    return preset || after || before;
+  }),
+
+  presetTimeKeys: Ember.computed(function() {
+    return _.keys(this.get("filterService.offerTimeRangePresets"));
+  }),
+
+  actions: {
+    redirectTofilters(queryParam) {
+      const offerFilter = {};
+      offerFilter[queryParam] = true;
+      this.get("router").transitionTo("offers_filters", {
+        queryParams: offerFilter
+      });
+    },
+
+    clearStateFilters() {
+      this.get("filterService").clearOfferStateFilters();
```

Comment:
I recently discovered that we can pass a `target` to the actions in `.hbs` files
Meaning we can do `{{ action 'enableReviewFilter' target=filterService}}` , which means we don't have to create those kinds of proxy actions.

Not a problem having them, just a good to know : )

Reply:
This is awesome !!